### PR TITLE
[TorchMhlo] turn on broadcast unittest

### PR DESCRIPTION
**Pull request by fortianyou**

### CODE REVIEW

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 1]
```diff
@@ -854,6 +893,65 @@ LogicalResult ConvertAtenOp<AtenDropoutOp>::matchAndRewrite(
...
+    AtenBroadcastToOp op,
+    OpAdaptor adaptor,
+    ConversionPatternRewriter& rewriter) const {
+  // Not a tensor type.
+  //
+  auto selfType = adaptor.self().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return op.emitError("Only tensor types are currently supported");
+
+  SmallVector<Value> dimsSize;
```
Comment (qiuxiafei) [inline pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:957]: nit: `_` connected variable name?

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 2]
```diff
@@ -854,6 +893,65 @@ LogicalResult ConvertAtenOp<AtenDropoutOp>::matchAndRewrite(
...
+        return op.emitError(
+            "For the new leading dimensions, the size cannot be set to -1.");
+      } else {
+        // Passing -1 as the size for a dimension means not changing the size
+        // of that dimension.
+        dsize = rewriter.create<tensor::DimOp>(
+            loc, adaptor.self(), d - leading_rank);
+      }
+    } else {
+      dsize = rewriter.create<ToI64Op>(loc, dsize).getResult();
```
Comment (qiuxiafei) [inline pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:981]: no need to consider `leading_rank` when it's not constant?

Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:981]: If the leading dimension `dsize` to expand is not a -1, we will use the `dsize` directly.

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp (hunk 1 above)
Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:957]: Indeed, LLVM standards prefer camel casing for the variables.

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp [hunk 3]
```diff
@@ -829,6 +848,26 @@ LogicalResult ConvertAtenOp<AtenLog2Op>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenSizeIntOp>::matchAndRewrite(
+    AtenSizeIntOp op,
+    OpAdaptor adaptor,
```
[the remark this answers is not in the document]
Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:854]: Ref to https://mlir.llvm.org/docs/OpDefinitions/#operand-adaptors

[the remark this answers is not in the document]
Comment (qiuxiafei) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:854]: Thanks ~

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp (hunk 1 above)
Comment (qiuxiafei) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp:953]: oops ... they even starts with upper case ... https://github.com/llvm/llvm-project/blob/169417531578501e955f12c79ecb8ff05333ae15/llvm/lib/TableGen/StringMatcher.cpp#L135

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**